src/AltinnCore/Runtime/RestControllers/DataController.cs

```diff
@@ -60,5 +168,270 @@ public async Task<ActionResult> Get(string org, string app, int instanceOwnerId,
+        /// <param name="elementType">identifies the data element type to create</param>
+        /// <returns>instance metadata with new data element</returns>
+        [Authorize]
+        [HttpPost]
+        [DisableFormValueModelBinding]
+        [RequestSizeLimit(REQUEST_SIZE_LIMIT)]
+        public async Task<ActionResult> CreateDataElement(
+            [FromRoute] string org,
+            [FromRoute] string app,
+            [FromRoute] int instanceOwnerId,
+            [FromRoute] Guid instanceGuid,
+            [FromQuery] string elementType = "default")
+        {
+            bool startService = true;
+
+            IServiceImplementation serviceImplementation = await PrepareServiceImplementation(org, app, elementType, startService);
+                
+            Application application = repositoryService.GetApplication(org, app);
+            if (application != null)
+            {
+                return NotFound($"AppId {org}/{app} was not found");
+            }
+
+            Instance instanceBefore = await instanceService.GetInstance(app, org, instanceOwnerId, instanceGuid);
+            if (instanceBefore == null)
+            {
+                return BadRequest("Unknown instance");
+            }       
+
+            object serviceModel = null;
+
+            if (Request.ContentType == null)
+            {
+               serviceModel = serviceImplementation.CreateNewServiceModel();
+            }
+            else
+            {
+                serviceModel = ParseContentAndDeserializeServiceModel(serviceImplementation.GetServiceModelType(), out ActionResult contentError);
+                if (contentError != null)
+                {
+                    return contentError;
+                }
+            }
+
+            serviceImplementation.SetServiceModel(serviceModel);
+
+            // send events to trigger application business logic
+            await serviceImplementation.RunServiceEvent(ServiceEventType.Instantiation);
+            await serviceImplementation.RunServiceEvent(ServiceEventType.ValidateInstantiation);
+
+            InstancesController.SetAppSelfLinks(instanceBefore, Request);
+           
+            Instance instanceAfter = await dataService.InsertData(serviceModel, instanceGuid, serviceImplementation.GetServiceModelType(), org, app, instanceOwnerId);
+
+            InstancesController.SetAppSelfLinks(instanceAfter, Request);
+            List<DataElement> createdElements = CompareAndReturnCreatedElements(instanceBefore, instanceAfter);           
+            string dataUrl = createdElements.First().DataLinks.Apps;
+
+            return Created(dataUrl, instanceAfter);
+        }
+
+        /// <summary>
+        /// Prepares the service implementation for a given dataElement, that has an xsd or json-schema.
+        /// </summary>
+        /// <param name="org">the organisation id</param>
+        /// <param name="app">the app name</param>
```

Comment:
Maybe use same explanation for "org" and "app", as done in the previous summaries.